University-advisor-web/Tools/GeoLocationApi.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace University_advisor_web.Tools
+{
+    public class GeoLocationApi
+    {
+        public string GetLocationJson(string url)
+        {
+            var request = (HttpWebRequest)WebRequest.Create(url);
+            try
+            {
+                var response = request.GetResponse();
+                using (var responseStream = response.GetResponseStream())
+                {
+                    var reader = new StreamReader(responseStream, Encoding.UTF8);
+                    return reader.ReadToEnd();
+                }
+            }
+            catch (WebException ex)
+            {
+                var errorResponse = ex.Response;
+                using (var responseStream = errorResponse.GetResponseStream())
+                {
+                    var reader = new StreamReader(responseStream, Encoding.GetEncoding("utf-8"));
+                    var errorText = reader.ReadToEnd();
+                    //Add logger
+                }
+                throw;
```

Comment:
Is necessary to rethrow exception?

Reply:
Not really, ill add logger and remove throw()